packages/experiments/src/components/Sidebar/Sidebar.tsx

```diff
@@ -0,0 +1,343 @@
+    }
+
+    const numActiveChildren = item.items.filter((child: ISidebarItemProps) => {
+      return child.active;
+    }).length;
+
+    const children = item.items.map((child: ISidebarItemProps) => {
+      child.onRender = this._renderSidebarButtonMenuItem;
+      return child;
+    });
+
+    if (item.name) {
+      children.unshift({
+        key: item.name + '-header',
+        name: item.name,
+        iconProps: { iconName: '' },
+        className: 'ba-SidebarContextualMenuButton-header ',
+        disabled: true,
+        styles: concatStyleSets(item.styles, {
+          root: {
+            borderBottomWidth: '1px',
+            borderBottomStyle: 'solid',
+            borderBottomColor: this._theme.semanticColors.bodyDivider
+          },
+          icon: {
+            width: '0',
+            marginRight: '0'
+          }
+        }),
+        onRender: this._renderSidebarButtonMenuItem
+      });
+    }
+
+    const ButtonAs = this._getButtonAs(item);
+
+    return (
+      <div key={item.key}>
+        <ButtonAs
+          key={item.key}
+          text={this.state.isCollapsed ? '' : item.name}
+          iconProps={item.iconProps}
+          menuIconProps={this.state.isCollapsed ? { iconName: '' } : item.subMenuIconProps}
+          menuProps={{
+            items: children,
+            directionalHint: DirectionalHint.rightTopEdge,
+            ariaLabel: item.name,
+            calloutProps: {
+              styles: {
+                root: {
+                  borderWidth: '0'
+                }
+              }
+            }
+          }}
+          menuTriggerKeyCode={KeyCodes.right}
+          className={this._getClassNames('ba-SidebarContextualMenuButton', item)}
+          role="menuitem"
+          ariaLabel={item.name}
+          title={item.title ? item.title : item.name}
+          styles={concatStyleSets(this._buttonStyles, item.styles)}
+          theme={this._theme}
+          checked={numActiveChildren > 0 ? true : false}
+          aria-current={numActiveChildren > 0 ? true : false}
+          disabled={item.disabled}
+          onClick={this._onItemClick(item)}
+        />
+      </div>
+    );
+  }
+
+  private _onItemClick(item: ISidebarItemProps): (ev: React.MouseEvent<HTMLButtonElement | HTMLAnchorElement>) => void {
+    return (ev: React.MouseEvent<HTMLButtonElement>): void => {
+      if (item.active) {
+        return;
+      }
+
+      if (item.onClick) {
+        item.onClick(ev, item);
+      }
+    };
+  }
+
+  // tslint:disable-next-line:no-any
+  private _getButtonAs(item?: ISidebarItemProps): any {
+    if (item && item.buttonAs) {
+      return item.buttonAs;
+    } else if (this.props.defaultButton) {
+      return this.props.defaultButton;
+    }
+
+    return SidebarButton;
+  }
+
+  private _getClassNames(defaultClass: string, item: ISidebarItemProps): string {
+    let className = defaultClass + ' ';
+    className += item.className ? item.className : '';
+    return className;
```

Comment:
Likely not worth changing because this implementation will change soon I imagine, but I find `return [className, item.className].join(' ')` a tad more robust.

![image](https://user-images.githubusercontent.com/706967/46551119-b8214e80-c88b-11e8-86be-ed2f9f3d2eaf.png)

Reply:
Feel free to point out anything you'd want to change. Someone will certainly being doing that work eventually. We just need to get this code into a place where they can do that.